### DISCUSSION TITLE
openid-client@5.1.4: The engine "node" is incompatible with this module. Expected version "^12.19.0 || ^14.15.0 || ^16.13.0". Got "17.2.0"

### DISCUSSION THREAD

**Discussion by 6xtvo** [nextauthjs/next-auth#4343] [Help] [answered]
### Environment

```
  System:
    OS: Windows 10 10.0.19043
    CPU: (6) x64 AMD Ryzen 5 4500U with Radeon Graphics
    Memory: 987.77 MB / 7.42 GB
  Binaries:
    Node: 17.2.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.3.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (100.0.1185.29)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    next: 12.1.0 => 12.1.0
    react: 17.0.2 => 17.0.2
   ```

### Reproduction URL

Not relevant, error was during the installation process

### Describe the issue

I've tried adding next-auth to my Next.js project and installed it using yarn by running `yarn add next-auth`, but it gives me the following error:
```
yarn add v1.22.17
[1/4] Resolving packages...
[2/4] Fetching packages...
error openid-client@5.1.4: The engine "node" is incompatible with this module. Expected version "^12.19.0 || ^14.15.0 || ^16.13.0". Got "17.2.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

### How to reproduce

1) Setup a **TypeScript** Next.js project using the method specified in https://nextjs.org/docs/basic-features/typescript. Ensure TypeScript has already been installed.
2) Run `yarn add next-auth`

Note: ensure you are using Node v17.2.0

### Expected behavior

The package should successfully install.

**Comment by ThangHuuVu** [comment, marked as answer]
We only support the mentioned Node versions at the moment. We're also working on replacing `openid-client`, so in the future, NextAuth.js won't be limited by the Node version.

**Comment by tego101** [comment]
How is this marked as answered and fixed? TO THIS DAY!!! I still have issues with yarn add next-auth

**Reply by 6xtvo** [reply]
`yarn add next-auth --ignore-engines`

**Reply by tego101** [reply]
Thank you.

**Reply by Zabavsky** [reply]
thank you 🙌

**Comment by commenda-main** [comment]
After installing `next-auth` with yarn I now I have to use the `--ignore-engines` flag every time I install anything.  Are there any updates on the plans to replace `openid-client`?

**Comment by amytych** [comment]
Just wanted to check in on this as it's also problematic for us. It's even more of an issue right now when `next-auth` specifies that Node 18 is supported:

https://github.com/nextauthjs/next-auth/blob/700daec919531f07c3b47d65d8d4363d076d65b8/packages/next-auth/package.json#L131

but it still depends on `openid-client` which is limited to Node 16:

https://github.com/nextauthjs/next-auth/blob/700daec919531f07c3b47d65d8d4363d076d65b8/packages/next-auth/package.json#L74

which produces the error when Node is 18:

```
error openid-client@5.1.9: The engine "node" is incompatible with this module. Expected version "^12.19.0 || ^14.15.0 || ^16.13.0". Got "18.12.1"
```

I wonder if this not an issue for folks, is everyone just using `--ignore-engines` flag when installing deps with `yarn`?

**Comment by mschipperheyn** [comment]
There is a new version of openid-client out. Any timeline on implementing this?

**Comment by monicaleep** [comment]
A workaround I found was that this was patched (`engines` were removed) in an openid-client version [v5.1.10](https://github.com/panva/node-openid-client/blob/v5.1.10/package.json), you can specify `yarn` to use that version in your `package.json`:
```json
 "resolutions": {
    "next-auth/openid-client": "5.1.10"
  }
```

**Reply by JohnGrisham** [reply]
Had to use openid-client instead of next-auth/openid-client then it worked for me.
